### PR TITLE
refactor(core): remove big switch from ReflectionCapabilities.factory

### PR DESCRIPTION
**Pull request by pgrzeszczak**
**What kind of change does this PR introduce?** (check one with "x")

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

**Other information**:
I refactored only TypeScript version because Dart doesn't have spread operator and doesn't provide access to an array of all arguments (as far as I know).

### CODE REVIEW

File: modules/@angular/core/src/reflection/reflection_capabilities.ts [hunk 1]
```diff
@@ -20,87 +20,10 @@ export class ReflectionCapabilities implements PlatformReflectionCapabilities {
...
-                a10: any, a11: any, a12: any, a13: any, a14: any, a15: any, a16: any, a17: any,
-                a18: any, a19: any, a20: any) =>
-                   new t(
-                       a1, a2, a3, a4, a5, a6, a7, a8, a9, a10, a11, a12, a13, a14, a15, a16, a17,
-                       a18, a19, a20);
-    };
-
-    throw new Error(
-        `Cannot create a factory for '${stringify(t)}' because its constructor has more than 20 arguments`);
+    if (t.length > 20)
```
Comment (vicb) [inline modules/@angular/core/src/reflection/reflection_capabilities.ts]: There is no need for this test any more

**Comment by vicb** [comment]
@tbosch @vsavkin what do you think ? Do you think the perf impact make it not mergeable ?

File: modules/@angular/core/src/reflection/reflection_capabilities.ts (hunk 1 above)
Comment (pgrzeszczak-neducatio) [inline modules/@angular/core/src/reflection/reflection_capabilities.ts]: I left this test in order to compability with Dart version

Comment (pgrzeszczak) [inline modules/@angular/core/src/reflection/reflection_capabilities.ts]: I left this test in order to compability with Dart version

**Comment by tbosch** [comment]
Let's wait with this until after final...

**Comment by vicb** [comment]
obsolete

**Pull request closed without merging**